### PR TITLE
Allow print_report to take report data & update fake report

### PR DESCRIPTION
**Pull request by DomAyre**
- Updates the print_report tool to accept report data as input which passes onto fetchSnpAttestation()
- Pass the hash of a test key as report data and get a fresh attestation, paste into the fake attestation report

### CODE REVIEW

**Comment by DomAyre** [comment]
System testing: https://github.com/microsoft/privacy-sandbox-dev/actions/runs/8207221010

File: scp/cc/azure/attestation/src/print_report.cc [hunk 1]
```diff
@@ -22,9 +22,11 @@ using google::scp::azure::attestation::fetchFakeSnpAttestation;
 using google::scp::azure::attestation::fetchSnpAttestation;
 using google::scp::azure::attestation::hasSnp;
 
-int main() {
-  const auto report =
-      hasSnp() ? fetchSnpAttestation() : fetchFakeSnpAttestation();
+int main(int argc, char* argv[]) {
+  std::string report_data = argc > 1 ? argv[1] : "";
+  const auto report = 
+    hasSnp() ? fetchSnpAttestation(report_data) : fetchFakeSnpAttestation();
```
Comment (beejones) [inline scp/cc/azure/attestation/src/print_report.cc:28]: fetchFakeSnpAttestation() should also pass in report_data with the fixed hash of our wellknown test key.

Comment (KenGordon) [inline scp/cc/azure/attestation/src/print_report.cc:28]: Did we reach a conclusion on "fake report needs to be checkable by CCF" vs "just stuff any old thing into the REPORT_DATA of a template report and don't worry that the signing is invalid"?

I think we went with the first and so would need a well known test key/hash that matches the fake report and then have the wrapping key be fixed on non SNP setups.

Comment (KenGordon) [reply scp/cc/azure/attestation/src/print_report.cc:28]: This particular path is just for printing a report so I don't think we ever care about providing report data to the fake one.  Maybe we should not allow this tool to run at all on non SNP hardware?

**Review by beejones — APPROVED** [review]

**Comment by DomAyre** [comment]
@KenGordon yeah Ronny and I had a call to clarify and we're happy with the fake report having the key hash baked in and the tests having the test key hardcoded

**Pull request merged**